pyro/primitives.py

```diff
@@ -92,15 +92,22 @@ class _Subsample(Distribution):
     Internal use only. This should only be used by `iarange`.
     """
 
-    def __init__(self, size, subsample_size, use_cuda=None):
+    def __init__(self, size, subsample_size, use_cuda=None, device=None):
         """
         :param int size: the size of the range to subsample from
         :param int subsample_size: the size of the returned subsample
         :param bool use_cuda: whether to use cuda tensors
```

Comment:
Maybe we can say
```
:param bool use_cuda: DEPRECATED, use `device` instead
```

Reply:
I have added that to `iarange` and `irange` docs since they'll be visible, but maybe I should mention it up front.

Reply:
oh that's fine, since those are the only public docs.